airbyte-integrations/connectors/source-stripe/unit_tests/test_streams.py

```diff
@@ -2,180 +2,178 @@
-            name="credit_notes",
-            path="credit_notes",
-            event_types=["credit_note.created", "credit_note.updated", "credit_note.voided"],
-            **args,
-        )
-
-    return mocker
-
-
-@pytest.fixture()
-def customers(stream_args):
-    def mocker(args=stream_args):
-        return IncrementalStripeStream(
-            name="customers",
-            path="customers",
-            use_cache=False,
-            event_types=["customer.created", "customer.updated"],
-            **args,
-        )
-
-    return mocker
-
-
-@pytest.fixture()
-def bank_accounts(customers, stream_args):
-    def mocker(args=stream_args):
-        return UpdatedCursorIncrementalStripeLazySubStream(
-            name="bank_accounts",
-            path=lambda self, stream_slice, *args, **kwargs: f"customers/{stream_slice[self.parent_id]}/sources",
-            parent=customers(),
-            event_types=["customer.source.created", "customer.source.expiring", "customer.source.updated"],
-            legacy_cursor_field=None,
-            parent_id="customer_id",
-            sub_items_attr="sources",
-            response_filter={"attr": "object", "value": "bank_account"},
-            extra_request_params={"object": "bank_account"},
-            record_extractor=FilteringRecordExtractor("updated", None, "bank_account"),
-            **args,
-        )
-
-    return mocker
-
+from source_stripe.streams import CheckoutSessionsLineItems, CustomerBalanceTransactions, Persons, SetupAttempts
 
-@pytest.fixture()
-def external_bank_accounts(stream_args):
-    def mocker(args=stream_args):
-        return UpdatedCursorIncrementalStripeStream(
-            name="external_account_bank_accounts",
-            path=lambda self, *args, **kwargs: f"accounts/{self.account_id}/external_accounts",
-            event_types=["account.external_account.created", "account.external_account.updated"],
-            legacy_cursor_field=None,
-            extra_request_params={"object": "bank_account"},
-            record_extractor=FilteringRecordExtractor("updated", None, "bank_account"),
-            **args,
-        )
 
-    return mocker
+def read_from_stream(stream, sync_mode, state):
+    records = []
+    for slice_ in stream.stream_slices(sync_mode=sync_mode, stream_state=state):
+        for record in stream.read_records(sync_mode=sync_mode, stream_slice=slice_, stream_state=state):
+            records.append(record)
+    return records
 
 
-def test_request_headers(accounts):
-    stream = accounts()
+def test_request_headers(stream_by_name):
+    stream = stream_by_name("accounts")
     headers = stream.request_headers()
     assert headers["Stripe-Version"] == "2022-11-15"
 
 
-def test_lazy_sub_stream(requests_mock, invoice_line_items, invoices, stream_args):
-    # First initial request to parent stream
-    requests_mock.get(
-        "https://api.stripe.com/v1/invoices",
-        json={
+bank_accounts_full_refresh_test_case = (
```

Comment:
This tests `Data not expanded when using the BankAccounts stream in the full refresh mode` 👍